product_supplierinfo_for_customer_sale/models/product_product.py

```diff
@@ -0,0 +1,27 @@
+# coding: utf-8
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import api, models
+
+
+class ProductProduct(models.Model):
+
+    _inherit = "product.product"
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=80):
+        res = super(ProductProduct, self).name_search(
+            name, args=args, operator=operator, limit=limit)
+        if res or not self._context.get('partner_id'):
+            return res
```

Comment:
We have this line uncovered from tests.
Add a new tests to validate that the default behaviour works!